container/container.go

```diff
@@ -667,15 +667,34 @@ func (container *Container) BuildCreateEndpointOptions(n libnetwork.Network, epC
 
 	if epConfig != nil {
 		ipam := epConfig.IPAMConfig
-		if ipam != nil && (ipam.IPv4Address != "" || ipam.IPv6Address != "" || len(ipam.LinkLocalIPs) > 0) {
+
+		if ipam != nil {
 			var ipList []net.IP
-			for _, ips := range ipam.LinkLocalIPs {
-				if ip := net.ParseIP(ips); ip != nil {
-					ipList = append(ipList, ip)
+			if len(ipam.LinkLocalIPs) > 0 {
+				for _, ips := range ipam.LinkLocalIPs {
+					if ip := net.ParseIP(ips); ip != nil {
+						ipList = append(ipList, ip)
+					} else {
+						return nil, fmt.Errorf("Error parsing parameter link-local-ip value(%s):%s", ipam.LinkLocalIPs, ips)
+					}
 				}
 			}
+			if ipam.IPv4Address != "" {
+
+				if ip := net.ParseIP(ipam.IPv4Address); ip == nil {
```

Comment:
given we are parsing again later for

`libnetwork.CreateOptionIpam(net.ParseIP(ipam.IPv4Address), net.ParseIP(ipam.IPv6Address), ipList, nil))`

I think we should define `var ip, ip6 net.IP`

and change these to 
```
if ip6 = net.ParseIP(ipam.IPv6Address); ip == nil &&  ipam.IPv6Address != "" {
                return nil, fmt.Errorf("Error parsing parameter ip value(%s)", ipam.IPv6Address)
}
```

and later change to 
```
`libnetwork.CreateOptionIpam(ip, ip6, ipList, nil))`